File: C++/2-Medium/00150._逆波兰表达式求值/00150._逆波兰表达式求值.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <string>
using namespace std;

int evalRPN(vector<string>& tokens) {
    stack<int> stk;
    for (string& token : tokens) {
        if (token == "+" || token == "-" || token == "*" || token == "/") {
            int num2 = stk.top(); stk.pop();
            int num1 = stk.top(); stk.pop();
            if (token == "+") stk.push(num1 + num2);
            else if (token == "-") stk.push(num1 - num2);
            else if (token == "*") stk.push(num1 * num2);
            else if (token == "/") stk.push(num1 / num2);
        } else {
            stk.push(stoi(token));
        }
    }
    return stk.top();
}

int main() {
    vector<string> tokens = {"2", "1", "+", "3", "*"};
    cout << "Result: " << evalRPN(tokens) << endl;
    return 0;
}

File: Python/2-Medium/00150._逆波兰表达式求值/00150._逆波兰表达式求值.py
from typing import List

def evaluate_reverse_polish_notation(tokens: List[str]) -> int:
    """
    计算逆波兰表达式的值。
    
    Args:
        tokens: 一个字符串列表，代表逆波兰表达式。
    
    Returns:
        表达式的计算结果。
    
    示例:
        >>> evaluate_reverse_polish_notation(["2", "1", "+", "3", "*"])
        9
        >>> evaluate_reverse_polish_notation(["4", "13", "5", "/", "+"])
        6
    """
    stack = []  # 初始化栈
    # 遍历每个元素
    for token in tokens:
        if token in "+-*/":  # 如果元素是运算符
            # 从栈中弹出两个元素作为操作数
            num2 = stack.pop()
            num1 = stack.pop()
            # 根据运算符进行计算，并将结果压回栈中
            if token == '+':
                stack.append(num1 + num2)
            elif token == '-':
                stack.append(num1 - num2)
            elif token == '*':
                stack.append(num1 * num2)
            else:  # 处理除法，注意向零取整
                stack.append(int(num1 / num2))
        else:  # 如果元素是数字，将其转换为整数后压入栈中
            stack.append(int(token))
    
    return stack[0]  # 遍历完成后，栈顶元素即为结果

# 运行示例
example_1 = ["2", "1", "+", "3", "*"]
example_2 = ["4", "13", "5", "/", "+"]
print(evaluate_reverse_polish_notation(example_1))  # 应输出 9
print(evaluate_reverse_polish_notation(example_2))  # 应输出 6